### PR TITLE
Fixes #26930 - support multi-repo composite publishing

### PR DESCRIPTION
**Pull request by jlsherrill**
this adds support for pulp3 for publishing a composite repo
with multiple components that contain the same repository

### CODE REVIEW

**Comment by jlsherrill** [comment]
TODO:

* [ ] write tests


To test:

1.  create a file repository and sync it, such as https://repos.fedorapeople.org/repos/pulp/pulp/fixtures/file/
2.  remove all files from it but 1 file, such as '1.iso'
3.  Publish the file repo in a content view named 'component1'
4.  resync the file repo
5.  remove all files from the file repo again except one, but leave a different file '2.iso'
6.  Publish the file repo in a content view named 'component2'
7.  create a new content view named 'composite', and make it a composite
8.  select both component1 and component2 and publish
9.  verify that both 1.iso and 2.iso are in the resulting content view.

File: app/lib/actions/pulp/repository/copy_all_units.rb [hunk 1]
```diff
@@ -2,16 +2,18 @@ module Actions
...
           rpm_filenames = options.fetch(:rpm_filenames, nil)
           solve_dependencies = options.fetch(:solve_dependencies, false)
 
-          plan_self(source_repo_id: source_repo.id,
-                    target_repo_id: target_repo.id,
-                    filter_ids: filter_ids,
-                    solve_dependencies: solve_dependencies,
-                    rpm_filenames: rpm_filenames)
+          source_repos.each do |source_repo|
+            plan_self(source_repo_id: source_repo.id,
```
Comment (sjha4) [inline app/lib/actions/pulp/repository/copy_all_units.rb]: We should check if we get a single repo here? I see one example here: https://github.com/Katello/katello/blob/c31deec0391d013f37b67364b9f5c0c859ad27bf/app/lib/actions/katello/repository/export.rb#L72

Comment (jlsherrill) [reply app/lib/actions/pulp/repository/copy_all_units.rb]: oh right, plan_self won't work in that case!  i think i'll need to refactor out the run phase to another action.

**Comment by sjha4** [comment]
Not sure I'm doing this right but with the steps above, the published composite content view seems to have 3 file units. Even the one that was deleted for both repos.

File: app/lib/actions/pulp/repository/copy_all_units.rb (hunk 1 above)
Comment (jlsherrill) [reply app/lib/actions/pulp/repository/copy_all_units.rb]: I updated this, but haven't tested my changes yet.  Will do so shortly

**Comment by sjha4** [comment]
@jlsherrill: There are conflicts on this..I will test this again to see if I am able to see the issue pre changes.. :smile:

**Comment by jlsherrill** [comment]
@sjha4 rebased!

**Pull request merged**